src/expression.d

```diff
@@ -2893,6 +2893,11 @@ extern (C++) abstract class Expression : RootObject
     final void checkDeprecated(Scope* sc, Dsymbol s)
     {
         s.checkDeprecated(loc, sc);
+        // Checks for deprecated and disabled have historically
+        // been in the same function, this is a temporary transition
+        // measure to keep the same semantic.
```

Comment:
Yeah it bothers me too. That is why I separated the [current function](https://github.com/dlang/dmd/blob/32c1d55fec6859e7d0d98344c01dbc694de0e7e9/src/dsymbol.d#L304-L346) into two functions, so at least one can be called without calling the other (which wasn't possible before). But in order not to introduce regressions the "high level" approach stays the same. In short, it just *enables* the possibility to call one without calling the other.